### PR TITLE
Add deploy branch make targets

### PR DESCRIPTION
**Pull request by sbrunner**

### CODE REVIEW

File: c2cgeoportal/scaffolds/update/CONST_Makefile_tmpl [hunk 1]
```diff
@@ -568,6 +574,23 @@ upgrade-db: alembic.ini alembic_static.ini $(VENV_BIN)/alembic$(PYTHON_BIN_POSTF
 	$(VENV_BIN)/alembic -c alembic.ini upgrade head
 	$(VENV_BIN)/alembic -c alembic_static.ini upgrade head
 
+# Deploy branch
+
+.PHONY: deploy-branch
+deploy-branch:
+	@read -p "Are you sure to deploy the branche (Y/n)?" RESP; RESP=`echo $$RESP | tr '[:upper:]' '[:lower:]'`; /usr/bin/test "$$RESP" == "y" -o "$$RESP" == "yes" -o "$$RESP" == ""
```
Comment (gberaudo) [inline c2cgeoportal/scaffolds/update/CONST_Makefile_tmpl]: typo: branch
What about printing the branch name?

File: c2cgeoportal/scaffolds/create/deploybranch.mk [hunk 2]
```diff
@@ -0,0 +1,5 @@
+INSTANCE_ID = $(GIT_BRANCH)
```
Comment (gberaudo) [inline c2cgeoportal/scaffolds/create/deploybranch.mk]: Do we need this .mk? Couldn't we directly pass the variables in the target?
If we need this, should it be renamed CONST_deploybranch.mk?

File: c2cgeoportal/scaffolds/update/CONST_Makefile_tmpl (hunk 1 above)
Comment (sbrunner) [inline c2cgeoportal/scaffolds/update/CONST_Makefile_tmpl]: or the destination directory name?

File: c2cgeoportal/scaffolds/create/deploybranch.mk (hunk 2 above)
Comment (sbrunner) [inline c2cgeoportal/scaffolds/create/deploybranch.mk]: Effectively we can pass the options in the target, I will have a look :-)

File: c2cgeoportal/scaffolds/update/CONST_Makefile_tmpl (hunk 1 above)
Comment (gberaudo) [inline c2cgeoportal/scaffolds/update/CONST_Makefile_tmpl]: It is longer but yes, it would make sense.

**Comment by sbrunner** [comment]
@gberaudo done :-)

File: c2cgeoportal/scaffolds/create/+package+.mk_tmpl [hunk 3]
```diff
@@ -6,7 +6,13 @@ VARS_FILES += ${VARS_FILE}
...
 
 APACHE_VHOST ?= {{apache_vhost}}
-INSTANCE_ID ?= {{project}}
+INSTANCE_ID ?= {{package}}
 TILECLOUD_CHAIN ?= FALSE
 
+# Deploy branch
+DEPLOY_BRANCH_DIR ?= /var/www/vhosts/$(APACHE_VHOST)/private/deploybranch
+GIT_REMOTE_URL ?= git@github.com:camptocamp/{{package}}.git
+DEPLOY_BRANCH_BASE_URL ?= https://example.com
```
Comment (gberaudo) [inline c2cgeoportal/scaffolds/create/+package+.mk_tmpl]: Shouldn' we fail here instead of using `https://example.com`?

Comment (sbrunner) [inline c2cgeoportal/scaffolds/create/+package+.mk_tmpl]: ?

File: c2cgeoportal/scaffolds/update/CONST_vars.yaml_tmpl [hunk 4]
```diff
@@ -275,7 +278,7 @@ vars:
     # The shortner base configuration
     shortener:
         # The base of created URL
-        base_url:  http://{host}{apache_entry_point}s/
+        base_url:  '{web_prorocol}://{host}{apache_entry_point}s/'
```
Comment (gberaudo) [inline c2cgeoportal/scaffolds/update/CONST_vars.yaml_tmpl]: typo

Comment (sbrunner) [inline c2cgeoportal/scaffolds/update/CONST_vars.yaml_tmpl]: Witch typo?

File: c2cgeoportal/scaffolds/update/CONST_vars.yaml_tmpl [hunk 5]
```diff
@@ -362,15 +365,12 @@ vars:
...
+          url: '{web_prorocol}://{host}/{instanceid}/wsgi'
           type: main
 
     # What web page is authorized to use the API
     authorized_referers:
-    - "http://{host}/{instanceid}"
-    - "https://{host}/{instanceid}"
-    - "http://{host}:8480/{instanceid}"
-    - "https://{host}:8480/{instanceid}"
+    - '{web_prorocol}://{host}/{instanceid}'
```
Comment (gberaudo) [inline c2cgeoportal/scaffolds/update/CONST_vars.yaml_tmpl]: Having both http and https is now deprecated?

File: c2cgeoportal/scaffolds/update/CONST_vars.yaml_tmpl (hunk 4 above)
Comment (sbrunner) [inline c2cgeoportal/scaffolds/update/CONST_vars.yaml_tmpl]: I just see

File: c2cgeoportal/scaffolds/update/CONST_vars.yaml_tmpl (hunk 5 above)
Comment (sbrunner) [inline c2cgeoportal/scaffolds/update/CONST_vars.yaml_tmpl]: Usually the application is on http or on https, we add booth to make it working on booth case, now with web_protcol we don't needs two entry for the booth case. it we should have booth we can add the manually :-)

**Comment by gberaudo** [comment]
Looks great to me.

**Comment by sbrunner** [comment]
Thanks @gberaudo :-)

**Pull request merged**